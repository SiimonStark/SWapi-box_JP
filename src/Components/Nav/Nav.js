import React from 'react';
import Button from '../Button/Button';

const Nav = () => {
  return(
    <nav>
      <Button name={'People'} />
      <Button name={'Planets'} />
      <Button name={'Vehicles'} />
      <Button name={'Favorites: 0'} />
    </nav>
  )
}

export default Nav;