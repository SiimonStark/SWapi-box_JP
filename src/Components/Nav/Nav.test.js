import React from 'react';
import {mount} from 'enzyme';
import Nav from './Nav';

let updateTabMock = jest.fn();

describe('Nav', ()=>{
  let wrapper;
  beforeEach(()=>{
    wrapper = mount(<Nav updateTab={updateTabMock}/>);
  });

  it('should match snapshot', ()=>{
    expect(wrapper).toMatchSnapshot();
  });

  it('should have children buttons', ()=>{
    expect(wrapper.find('nav').children().length).toEqual(4);
  })
});