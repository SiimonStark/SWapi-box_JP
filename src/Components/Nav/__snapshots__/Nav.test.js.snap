// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
<Nav
  updateTab={[MockFunction]}
>
  <nav>
    <Button
      name="People"
      updateTab={[MockFunction]}
    >
      <button
        id="People"
        onClick={[MockFunction]}
      >
        People
      </button>
    </Button>
    <Button
      name="Planets"
      updateTab={[MockFunction]}
    >
      <button
        id="Planets"
        onClick={[MockFunction]}
      >
        Planets
      </button>
    </Button>
    <Button
      name="Vehicles"
      updateTab={[MockFunction]}
    >
      <button
        id="Vehicles"
        onClick={[MockFunction]}
      >
        Vehicles
      </button>
    </Button>
    <Button
      name="Favorites"
      updateTab={[MockFunction]}
    >
      <button
        id="Favorites"
        onClick={[MockFunction]}
      >
        Favorites
      </button>
    </Button>
  </nav>
</Nav>
`;
